### PR TITLE
kubelet/rkt: add created/started/failed/killing events.

### PR DESCRIPTION
**Pull request by yifan-gu**
Also clear container reference when killing a container.

But we still need the gc to clean the readiness, reference for containers that exit voluntarily. Will file an issue for this.

This depends on #https://github.com/kubernetes/kubernetes/pull/13041

### CODE REVIEW

**Comment by yifan-gu** [comment]
@yujuhong Just rebased, feel free to review :)

File: pkg/kubelet/rkt/rkt.go [hunk 1]
```diff
@@ -614,21 +618,49 @@ func (r *runtime) preparePod(pod *api.Pod, pullSecrets []api.Secret) (string, er
...
+		id := string(c.ID)
+		ref, ok := r.containerRefManager.GetRef(id)
+		if !ok {
+			glog.Warningf("No ref for container %q", id)
+			continue
+		}
+
+		switch reason {
+		case "created":
+			r.recorder.Eventf(ref, "created", "Created with %q, unit file %q", id, unit)
```
Comment (dchen1107) [inline pkg/kubelet/rkt/rkt.go]: Reason should be CamelCase. Also please align the reasons you listed here with what we had done for docker container runtime:

https://github.com/kubernetes/kubernetes/blob/a945785409d5b68f3a2721d2209300edb5abf1ce/pkg/kubelet/dockertools/manager.go#L671

So Created reason means the same thing no matter which runtime is enabled. Same for rest of reasons in events.

Comment (yifan-gu) [inline pkg/kubelet/rkt/rkt.go]: Ok, updated.

**Comment by yifan-gu** [comment]
@yujuhong Addressed @dchen1107 's comments. PTAL, thx :)

**Comment by yujuhong** [comment]
@yifan-gu, sorry I was oncall the past two days, and was mostly at talks today. Reviewing it now.

**Comment by yujuhong** [comment]
> But we still need the gc to clean the readiness, reference for containers that exit voluntarily. Will file an issue for this.

Good point. I don't think we do that (gc the readiness) for docker. Perhaps we should make them expire since they should be check and update periodically.

**Comment by yujuhong** [comment]
LGTM with one question

**Comment by yifan-gu** [comment]
Added a doc explaining why take first 8 characters as the shortened version of the rkt id.

> @yifan-gu, sorry I was oncall the past two days, and was mostly at talks today. Reviewing it now.

No worries, it's just a friendly ping in case this pr is buried :) BTW if you have any PR needs me to help review, feel free to ping me.

**Comment by yujuhong** [comment]
> No worries, it's just a friendly ping in case this pr is buried :) BTW if you have any PR needs me to help review, feel free to ping me.

I'll take up the offer sometime next week :)

LGTM.

**Comment by brendandburns** [comment]
closing and re-opening to ping shippable

**Comment by brendandburns** [comment]
@k8s-bot ok to test

	pr builder appears to be missing, activating due to 'lgtm' label.

**Pull request merged**